react/features/base/devices/middleware.js

```diff
@@ -28,6 +30,12 @@ MiddlewareRegistry.register(store => next => action => {
     case SET_VIDEO_INPUT_DEVICE:
         APP.UI.emitEvent(UIEvents.VIDEO_DEVICE_CHANGED, action.deviceId);
         break;
+    case SHOW_DEVICE_ERROR:
+        store.dispatch(openDialog(DeviceErrorDialog, {
+            micError: action.micError,
+            cameraError: action.cameraError
+        }));
+        break;
```

Comment:
Is it preferred to open the dialog in the middleware instead of the action calling openDialog?